src/EnergyPlus/Material.cc

```diff
@@ -2756,6 +2758,85 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
+}
+
+void GetVariableAbsorptanceInput(EnergyPlusData &state, bool &errorsFound)
+{
+    int IOStat; // IO Status when calling get input subroutine
+    int numAlphas;
+    int numNumbers;
+    Array1D_string alphas(7);   // character string data
+    Array1D<Real64> numbers(1); // numeric data
+    std::string_view cCurrentModuleObject{"MaterialProperty:VariableAbsorptance"};
+    int numVariAbs = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
+    for (int i = 1; i <= numVariAbs; ++i) {
+        // Call Input Get routine to retrieve material data
+        state.dataInputProcessing->inputProcessor->getObjectItem(state,
+                                                                 cCurrentModuleObject,
+                                                                 i,
+                                                                 alphas,
+                                                                 numAlphas,
+                                                                 numbers,
+                                                                 numNumbers,
+                                                                 IOStat,
+                                                                 state.dataIPShortCut->lNumericFieldBlanks,
+                                                                 state.dataIPShortCut->lAlphaFieldBlanks,
+                                                                 state.dataIPShortCut->cAlphaFieldNames,
+                                                                 state.dataIPShortCut->cNumericFieldNames);
+
+        // Load the material derived type from the input data.
+        int MaterNum = UtilityRoutines::FindItemInPtrList(alphas(2), state.dataMaterial->Material);
+        if (MaterNum == 0) {
+            ShowSevereError(state,
+                            format("{}: invalid {} entered={}, must match to a valid Material name.",
+                                   cCurrentModuleObject,
+                                   state.dataIPShortCut->cAlphaFieldNames(1),
+                                   alphas(2)));
+            errorsFound = true;
+            return;
+        }
+        auto *thisMaterial = state.dataMaterial->Material(MaterNum);
+
+        if (thisMaterial->Group != Material::MaterialGroup::RegularMaterial) {
+            ShowSevereError(
+                state,
+                format("{}: Reference Material is not appropriate type for Thermal/Solar Absorptance properties, material={}, must have regular "
+                       "properties (Thermal/Solar Absorptance)",
+                       cCurrentModuleObject,
+                       thisMaterial->Name));
+            errorsFound = true;
+            return;
+        }
+
+        Material::VariableAbsCtrlSignal controlSignal = Material::VariableAbsCtrlSignal::SurfaceTemperature; // default value
```

Comment:
Shouldn't this be `thisMaterial->absorpVarCtrlSignal =`? Don't you have to save this in `thisMaterial`?

Reply:
yes, I do need to save it. forgot to add it back after removing temp variables